app/src/main/java/com/bogdan/codeforceswatcher/features/actions/models/ActionItem.kt

```diff
@@ -0,0 +1,61 @@
+package com.bogdan.codeforceswatcher.features.actions.models
+
+import android.text.SpannableString
+import android.text.SpannableStringBuilder
+import android.text.style.UnderlineSpan
+import androidx.core.text.HtmlCompat
+import com.bogdan.codeforceswatcher.CwApp
+import com.bogdan.codeforceswatcher.R
+import com.bogdan.codeforceswatcher.features.users.models.colorTextByUserRank
+import org.ocpsoft.prettytime.PrettyTime
+import java.util.*
+
+class ActionItem(action: CFAction) {
+
+    lateinit var commentatorHandle: CharSequence
+    lateinit var title: CharSequence
+    lateinit var content: CharSequence
+    lateinit var commentatorAvatar: String
+    lateinit var timeAgo: String
+
+    init {
+        if (action.comment != null) {
+            this.commentatorAvatar = (if (action.comment.commentatorAvatar.startsWith("https:")) {
+                action.comment.commentatorAvatar
+            } else {
+                "https:${action.comment.commentatorAvatar}"
+            })
+
+            this.commentatorHandle = formatCommentatorHandle(
+                action.comment.commentatorHandle, action.comment.commentatorRank
+            )
+
+            this.title = underlineTitle(action.blogEntry.title)
+            this.timeAgo = PrettyTime().format(Date(action.comment.creationTimeSeconds * 1000))
+            this.content = convertFromHtml(action.comment.text)
+        }
+    }
+
+    private fun formatCommentatorHandle(handle: String, rank: String?): CharSequence {
+        val colorHandle = colorTextByUserRank(handle, rank, CwApp.app)
+        val commentedByString = CwApp.app.getString(R.string.commented_by)
+        val handlePosition = commentedByString.indexOf("%1\$s")
+
+        return SpannableStringBuilder(commentedByString)
+            .replace(handlePosition, handlePosition + "%1\$s".length, colorHandle)
+    }
+
+    private fun underlineTitle(title: String): SpannableString {
```

Comment:
Just a question. Don't you think that in Codeforces Watcher underline looks strange? Won't it be better without it?